### PR TITLE
Update request body from request param to json for lease endpoint

### PR DESCRIPTION
**Pull request by nitinprabhu**
- Existing request body is empty even if the HTTP request type was post.

- Updated request body to pass parameters as Json instead of request param.

- Feign Client used in `service-auth-provider-java-client` can now invoke passing micro service name and otppassword in request body as json.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/auth/provider/service/api/auth/AuthController.java [hunk 1]
```diff
@@ -25,10 +28,9 @@ public AuthController(AuthService authService) {
     @PostMapping("/lease")
     @ResponseBody
     public ResponseEntity<?> lease(
-        @RequestParam @NotEmpty String microservice,
-        @RequestParam @NotEmpty String oneTimePassword
+        @Valid @RequestBody TokenDetails tokenDetails
     ) {
-        String token = authService.lease(microservice, oneTimePassword);
+        String token = authService.lease(tokenDetails.microservice, tokenDetails.oneTimePassword);
```
Comment (rkondratowicz) [inline api/src/main/java/uk/gov/hmcts/auth/provider/service/api/auth/AuthController.java]: let's support both old and new endpoint

File: api/src/main/java/uk/gov/hmcts/auth/provider/service/api/model/TokenDetails.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.auth.provider.service.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.hibernate.validator.constraints.NotEmpty;
+
+public class TokenDetails {
```
Comment (rkondratowicz) [inline api/src/main/java/uk/gov/hmcts/auth/provider/service/api/model/TokenDetails.java]: Those are not token details :)
Perhaps `SignIn`?

File: api/src/test/java/uk/gov/hmcts/auth/provider/service/api/componenttests/dsl/ServiceAuthProviderTestDsl.java [hunk 3]
```diff
@@ -109,4 +116,12 @@ public ServiceAuthProviderThenDsl and() {
         return objectMapper.readValue(resultActions.andReturn().getResponse().getContentAsByteArray(), valueType);
     }
 
+    private String createJson(final String microservice, final String password) throws IOException {
+        Map<String, String> tokenDetails = ImmutableMap.of(
+            "microservice", microservice,
+            "oneTimePassword", password
+        );
+
+        return new ObjectMapper().writeValueAsString(tokenDetails);
```
Comment (rkondratowicz) [inline api/src/test/java/uk/gov/hmcts/auth/provider/service/api/componenttests/dsl/ServiceAuthProviderTestDsl.java]: why not just serialize the model?

File: token-generator/src/main/java/uk/gov/hmcts/auth/provider/service/token/HttpComponentsBasedServiceTokenGenerator.java [hunk 4]
```diff
@@ -36,11 +39,19 @@ public String generate() {
...
-                new BasicNameValuePair("microservice", microservice),
-                new BasicNameValuePair("oneTimePassword", oneTimePassword)
-            )));
+            request.setEntity(
+                new StringEntity(
+                    new ObjectMapper()
+                        .writeValueAsString(tokenDetails)
+                ));
+
+            request.addHeader("content-type", "application/json");
```
Comment (rkondratowicz) [inline token-generator/src/main/java/uk/gov/hmcts/auth/provider/service/token/HttpComponentsBasedServiceTokenGenerator.java]: `Content-Type`
Or even better use constants from a library. (I think apache commons is already included and has it)

File: api/src/main/java/uk/gov/hmcts/auth/provider/service/api/model/TokenDetails.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    @NotEmpty
+    public final String microservice;
+
+    @NotEmpty
+    public final String oneTimePassword;
+
+    public TokenDetails(
+        @JsonProperty("microservice") String microservice,
+        @JsonProperty("oneTimePassword") String oneTimePassword
```
Comment (rkondratowicz) [inline api/src/main/java/uk/gov/hmcts/auth/provider/service/api/model/TokenDetails.java]: should be `one_time_password` according to hmcts standards:
https://hmcts.github.io/restful-api-standards/#118

**Comment by nitinprabhu** [comment]
These changes are not required anymore.Hence closing this PR.

**Pull request closed without merging**